### ISSUE TITLE
Corn vanishes, or goes from rotten to old or normal.

### ISSUE THREAD

**Issue by Lamandus** [CleverRaven/Cataclysm-DDA#47338]
# Describe the bug
I collected some corn from a silo. Some of which turned rotten, so I disposed it outside. After a good night sleep it got edible again.

# Steps To Reproduce

1. get rotten corn.
2. wait or go to sleep
3. it gets either old or "normal" or it vanishes

# Expected behavior
Stay rotten.

# Screenshots
![grafik](https://user-images.githubusercontent.com/33199510/107230667-a2099700-6a1f-11eb-91d7-661ec3ac140f.png)
![grafik](https://user-images.githubusercontent.com/33199510/107231754-f3feec80-6a20-11eb-960c-b21461b78a5f.png)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-9473-g83b0843-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    meins [mine],
    Nonperishable Overhaul [Nonperishable_Overhaul]
]

# Additional context
It is very hard to discribe, but the corn changes its behaviour from load to load. If I get more info, I keep you informed.

**Comment by PaulBlay** [comment]
Does it just lose the 'rotten' or does it go 'fresh'? I remember a bug a while back where rotten milk would turn fresh when you 'r'eloaded containers with it. (That got fixed, IIRC)

**Comment by Lamandus** [comment]
sometimes it does, sometimes it don't. I think the current weather in Sri Lanka is also a factor.

**Comment by anothersimulacrum** [comment]
Can you reproduce it without `Nonperishable Overhaul [Nonperishable_Overhaul]`?

**Comment by Hirmuolio** [comment]
Something is wrong with time.
The items on ground and most items in general seem to be ftom about 70 days from the future.

Looking at the save it seems the you started and are currently living in time before the cataclysm started.  
According to the save file 
Start of cataclysm: 18662400
Game start: 12470400

The game should never start before the cataclysm. Something must have gone wrong in the world/character/scenario creation.

**Comment by PaulBlay** [comment]
Not making any accusations, because I've done it, but did @Lamandus mess with the save files at all? Like copying over a character from an earlier game to a new map?

**Comment by Lamandus** [comment]
> 
> 
> Not making any accusations, because I've done it, but did @Lamandus mess with the save files at all? Like copying over a character from an earlier game to a new map?

nope. clean save. I even didn't edit the mods.json (I do sometimes tho).

**Comment by Hirmuolio** [comment]
I may have found the problem.

If the start day is set to -1 (random) it is set here for `calendar::start_of_cataclysm` https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L12215

But for the scenario it is set here and there is no handling at all for random start day. So "random" day would result in starting day of -1. https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/scenario.cpp#L115

So few lines later here https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L12221
The `scen->initial_day()` used for `calendar::start_of_game` will be different from initial_days used for `calendar::start_of_cataclysm` if the initial day is random.

Also if the scenario has defined starting day https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/scenario.cpp#L105
If the scenario `"day"` is smaller than user defined `"INITIAL_DAY"` then result would be same issue and game would start at time before the cataclysm (this day property is not used on any scenario).

The easiest fix would be to remove the option to set the day in scenario. It isn't used anywhere anyways.

**Comment by Hirmuolio** [comment]
Trying to think of an elegant fix. Part of it is easy and part of it is vague on how things should work.

What is the scenario specific date meant to be?
Is it supposed to be "start at this time of year" (for example start on third day of winter)?
Is it supposed to be "start this much later" (for example start on second year after cataclysm)?
Both of these are used for scenarios.
Both of these can't exist without leaving annoying edge cases in.

The second problem is how "start of cataclysm" and "start of game" are tied together in the same "INITIAL_DAY" option.

I think best fix would be to split "INITIAL_DAY" into two: "day_of_cataclysm" and "days_after_cataclysm" options.
The scenario would keep its "start at this time of year" and get another "days_after_cataclysm" property.

In both of these cases "days_after_cataclysm" would be minimum number of days that has passed since cataclysm.

In this system `calendar::start_of_cataclysm` would be same `calendar::turn_zero + start_of_cataclysm`

`calendar::start_of_game` would be a bit different. It would be something like this:
```
"days_after_cataclysm" = max( "days_after_cataclysm"-option, "days_after_cataclysm"-scenario_property )
if "start at this time of year"
    calendar::start_of_game = calendar::start_of_cataclysm + "start at this time of year"
    if calendar::start_of_game < calendar::start_of_cataclysm + "days_after_cataclysm"
        calendar::start_of_game = calendar::start_of_game + 1 year
else
    calendar::start_of_game = calendar::start_of_cataclysm + "days_after_cataclysm"
```

This way it is possible to define scenario that starts at least 1 year after cataclysm on winter,  scenario that starts half year after cataclysm, and scenario that starts on first autumn after cataclysm.

This would probably have to wait until after 0.F release. Until then some hotfix will be enough.

**Comment by Hirmuolio** [comment]
Extra annoyance:
"day" is not safe time definition.
Day 25 of summer can be day 1 of spring of next year with certain season lengths.

Some season length agnostic date system would be very useful.
